htlcswitch/switch.go

```diff
@@ -349,6 +356,38 @@ func (s *Switch) ProcessContractResolution(msg contractcourt.ResolutionMsg) erro
 	return nil
 }
 
+
+func hash(s string) uint32 {
+        h := fnv.New32a()
+        h.Write([]byte(s))
+        return h.Sum32()
+}
+
```

Comment:
@vibhaa I added a few comments to describe what is being done here. The main primitive provided by firebase is key : value, so reusing it for other statistics would basically involve finding a unique key. For switches, I did it in a slightly hacky (but I think should be correct) way here since switches did not have a unique id already. So something similar could presumably done at other places too. But for instance when collecting queue length statistics, queue.go / or link.go do not know which switch they belong to - so that would be an issue. I was thinking of just exposing the queue length information to the switches just for debugging purposes, but we could remove that if we want to get it into lnd eventually.

Also, the visualization was being processed in python - should I add those python files in the repo as well / or I can share it separately since it doesn't have any lnd significance?

Reply:
also, there are a few issues with this - it might not need a fix yet, so I will just open issues about them (?) 
 - this just supports visualizing a currently running process -- but data isn't deleted automatically, so next processes wouldn't run. 
 - we don't use authentication for storing the results online

Reply:
Re: the issues, I'd just open issues (not sure I follow the first point though).

The hack for switches seems fine and I agree, maybe we can find something of that sort for other statistics too, but not sure I follow the remaining stuff on queue.go/link.go not knowing what they belong to. Is there no self field that helps identify the switch? Is that what you're saying? If so, how does queue length information exposing help?